lib/widgets/upload_button.dart

```diff
@@ -0,0 +1,111 @@
+import 'dart:typed_data';
+
+import 'package:acs_upb_mobile/generated/l10n.dart';
+import 'package:file_picker/file_picker.dart';
+import 'package:flutter/material.dart';
+import 'package:positioned_tap_detector/positioned_tap_detector.dart';
+
+class UploadButtonController {
+  UploadButtonController({this.onUpdate});
+
+  _UploadButtonState _uploadButtonState;
+  Uint8List newUploadedImageBytes;
+  void Function() onUpdate;
+
+  Uint8List get uploadImageBytes => newUploadedImageBytes;
+
+  void setNewImg(Uint8List uploadedImageBytes) {
+    if (_uploadButtonState == null) return;
+    newUploadedImageBytes = uploadedImageBytes;
+    onUpdate();
+  }
+}
+
+// Returns a widget that behaves similarly to a textfield with a "clear"
+// button, except it actually allows the user to select an image from the
+// gallery instead of inputting text directly.
+class UploadButton extends StatefulWidget {
+  const UploadButton(this.imageFieldController,
```

Comment:
We don't actually need to pass the TextEditingController as a parameter. It can just be initialized in the state (move the property to the state and initialize it with TextEditingController() directly).